pkg/webhook/cm/plugins/sloconfig/resource_qos_checker.go

```diff
@@ -111,6 +115,254 @@ func CheckNetQosCfg(cfg *configuration.ResourceQOSCfg, fldPath *field.Path) erro
+				egressRequestTotal -= cfg.ClusterStrategy.BEClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+
+		if ingressRequestTotal < 0 || egressRequestTotal < 0 {
+			return fmt.Errorf("ClusterStrategy total net bandwidth percentage already over 100")
+		}
+	}
+
+	// check netqos config is valid for each node strategy
+	for idx, nodeStrategy := range cfg.NodeStrategies {
+		ingressRequestTotal, egressRequestTotal := 100, 100
+		if nodeStrategy.LSRClass != nil && nodeStrategy.LSRClass.NetworkQOS != nil {
+			if nodeStrategy.LSRClass.NetworkQOS.IngressRequest != nil && nodeStrategy.LSRClass.NetworkQOS.IngressRequest.Type == intstr.Int {
+				ingressRequestTotal -= nodeStrategy.LSRClass.NetworkQOS.IngressRequest.IntValue()
+			}
+			if nodeStrategy.LSRClass.NetworkQOS.EgressRequest != nil && nodeStrategy.LSRClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= nodeStrategy.LSRClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+		if nodeStrategy.LSClass != nil && nodeStrategy.LSClass.NetworkQOS != nil {
+			if nodeStrategy.LSClass.NetworkQOS.IngressRequest != nil && nodeStrategy.LSClass.NetworkQOS.IngressRequest.Type == intstr.Int {
+				ingressRequestTotal -= nodeStrategy.LSClass.NetworkQOS.IngressRequest.IntValue()
+			}
+			if nodeStrategy.LSClass.NetworkQOS.EgressRequest != nil && nodeStrategy.LSClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= nodeStrategy.LSClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+		if nodeStrategy.BEClass != nil && nodeStrategy.BEClass.NetworkQOS != nil {
+			if nodeStrategy.BEClass.NetworkQOS.IngressRequest != nil && nodeStrategy.BEClass.NetworkQOS.IngressRequest.Type == intstr.Int {
+				ingressRequestTotal -= nodeStrategy.BEClass.NetworkQOS.IngressRequest.IntValue()
+			}
+			if nodeStrategy.BEClass.NetworkQOS.EgressRequest != nil && nodeStrategy.BEClass.NetworkQOS.EgressRequest.Type == intstr.Int {
+				egressRequestTotal -= nodeStrategy.BEClass.NetworkQOS.EgressRequest.IntValue()
+			}
+		}
+		if ingressRequestTotal < 0 || egressRequestTotal < 0 {
+			return fmt.Errorf("cfg.NodeStrategies[%d] total net bandwidth percentage already over 100", idx)
+		}
+	}
+
+	return nil
+}
+
+func CheckTotalQuantityIsValid(cfg *configuration.ResourceQOSCfg, sysCfg *configuration.SystemCfg) error {
+	if cfg == nil {
+		return nil
+	}
+
+	// check netqos config is valid for cluster strategy
+	if cfg.ClusterStrategy != nil && sysCfg.ClusterStrategy != nil {
+		ingressRequestTotal, egressRequestTotal := sysCfg.ClusterStrategy.TotalNetworkBandwidth, sysCfg.ClusterStrategy.TotalNetworkBandwidth
+		if cfg.ClusterStrategy.LSRClass != nil && cfg.ClusterStrategy.LSRClass.NetworkQOS != nil {
+			if cfg.ClusterStrategy.LSRClass.NetworkQOS.IngressRequest != nil && cfg.ClusterStrategy.LSRClass.NetworkQOS.IngressRequest.Type == intstr.String {
+				ingressRequestTotal.Sub(resource.MustParse(cfg.ClusterStrategy.LSRClass.NetworkQOS.IngressRequest.StrVal))
```

Comment:
Here,It's not recommend to use `resource.MustParse` since it can generate panic.  And the codes looks the same as above codes but just different total and format..

Reply:
done